hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -585,11 +586,15 @@ protected static long sumBlockLengths(OmKeyInfo omKeyInfo) {
   /**
    * Return bucket info for the specified bucket.
    */
+  @Nullable
   protected OmBucketInfo getBucketInfo(OMMetadataManager omMetadataManager,
       String volume, String bucket) {
-    return omMetadataManager.getBucketTable().getCacheValue(
-        new CacheKey<>(omMetadataManager.getBucketKey(volume, bucket)))
-        .getCacheValue();
+    String bucketKey = omMetadataManager.getBucketKey(volume, bucket);
+
+    CacheValue<OmBucketInfo> value = omMetadataManager.getBucketTable()
+        .getCacheValue(new CacheKey<>(bucketKey));
+
+    return value != null ? value.getCacheValue() : null;
```

Comment:
When is OmBucketInfo null? Does our code currently handle the OmBucketInfo null case?

Reply:
When the given `bucketKey` is not exist in cache, or the cached value is `null`.
The original code will throw NPE at `value.getCacheValue()` in the first case.

Reply:
In `TestOMOpenKeysDeleteRequest#testDeleteOpenKeysNotInTable()`,
the bucket info of the open keys were not added to table.
`getBucketInfo` will return null.

Reply:
Ideally this method would do a normal key read and not strictly a cache read. The result would be the same but it would no longer depend on the bucket table being a full cache, which is an implementation detail outside of this class.